tests/test_hashring.py

```diff
@@ -14,24 +14,21 @@ def __repr__(self):
         return f"<Node {self.id}>"
 
 
-class HashRingTest(unittest.TestCase):
-    def setUp(self):
-        self.node0 = Node(0)
-        self.node1 = Node(1)
-        self.node2 = Node(2)
+@pytest.fixture
+def hash_ring():
+    return HashRing([Node(i) for i in range(3)])
 
-        self.nodes = [self.node0, self.node1, self.node2]
-        self.ring = HashRing(self.nodes)
 
-    def test_hashring(self):
-        ids = []
+def test_hashring(hash_ring):
```

Comment:
The `hash_ring` fixture is requested here and references are adjusted in the test function.